compiler/passes/src/type_checking/check_expressions.rs

```diff
@@ -613,11 +636,16 @@ impl<'a> ExpressionVisitor<'a> for TypeChecker<'a> {
 
     fn visit_tuple(&mut self, input: &'a TupleExpression, expected: &Self::AdditionalInput) -> Self::Output {
         match input.elements.len() {
-            0 => Some(self.assert_and_return_type(Type::Unit, expected, input.span())),
-            1 => self.visit_expression(&input.elements[0], expected),
+            0 => {
```

Comment:
It looks like this may not be reachable, given that we check for the empty tuple in `visit_expression`?

Reply:
Good catch!